server/platform/services/sharedchannel/sync_recv.go

```diff
@@ -79,7 +80,17 @@ func (scs *Service) processSyncMessage(c request.CTX, syncMsg *model.SyncMsg, rc
 		return fmt.Errorf("cannot check channel share state for sync message: %w", err)
 	}
 	if !exists {
-		return fmt.Errorf("cannot process sync message; channel not shared with remote: %w", ErrRemoteIDMismatch)
+		// Update the UI to ensure the link icon is removed
+		scs.notifyClientsForSharedChannelUpdate(targetChannel)
+
+		// Return both errors to maintain backward compatibility and enable new behavior:
```

Comment:
In this case, we were not managing the errors in any way on the remote that originated the message, so I don't think we need to maintain backwards compatibility. If an old server receives the new error, it will log a better error message but will not react to it.

Reply:
good to know, currently erring on the safe side, will remove the previous error message.